workflows/templates/ncn-reboot-common.yaml

```
@@ -0,0 +1,305 @@
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{inputs.parameters.dryRun}}"
+                - name: mediaHost
+                  value: "{{inputs.parameters.media_host}}"                         
+                - name: scriptContent
+                  value: |
+                    TARGET_NCN="{{inputs.parameters.targetNcn}}"
+                    TARGET_NCN_mgmt_host="${TARGET_NCN}-mgmt"
+                    # Get Keycloak token
+                    CLIENT_SECRET=$(kubectl get secret admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d)
+                    TOKEN=$(curl -k -s -S -d grant_type=client_credentials \
+                      -d client_id=admin-client \
+                      -d client_secret="$CLIENT_SECRET" \
+                      https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
+
+                    # Get xnames
+                    TARGET_XNAME=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" \
+                      "https://api-gw-service-nmn.local/apis/sls/v1/search/hardware?extra_properties.Role=Management" |
+                      jq -r ".[] | select(.ExtraProperties.Aliases[] | contains(\"${TARGET_NCN}\")) | .Xname")
+
+                    TARGET_MGMT_XNAME=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" \
+                      "https://api-gw-service-nmn.local/apis/sls/v1/search/hardware?extra_properties.Role=Management" |
+                      jq -r ".[] | select(.ExtraProperties.Aliases[] | contains(\"${TARGET_NCN}\")) | .Parent")
+
+                    # Get IPMI credentials
+                    IPMI_USERNAME=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" \
+                      "https://api-gw-service-nmn.local/apis/scsd/v1/bmc/creds" |
+                      jq -r ".Targets[] | select(.Xname == \"$TARGET_MGMT_XNAME\") | .Username")
+
+                    IPMI_PASSWORD=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" \
+                      "https://api-gw-service-nmn.local/apis/scsd/v1/bmc/creds" |
+                      jq -r ".Targets[] | select(.Xname == \"$TARGET_MGMT_XNAME\") | .Password")
+
+                    # Power functions
+                    check_power_status() {
+                      output=$(ipmitool -I lanplus -U "${IPMI_USERNAME}" -P "${IPMI_PASSWORD}" -H "${TARGET_NCN_mgmt_host}" power status 2>&1)
+                      if [[ $? -ne 0 ]]; then
+                        echo "ERROR Failed getting power status $TARGET_NCN"
+                        exit 1
+                      else
+                        echo "$output" | awk '{print $4}'
+                      fi
+                    }
+
+                    echo "Powering off ${TARGET_NCN_mgmt_host}"
+                    ipmitool -I lanplus -U "${IPMI_USERNAME}" -P "${IPMI_PASSWORD}" -H "${TARGET_NCN_mgmt_host}" power off
+                    if [[ $? -ne 0 ]]; then
```

You can check the command exit value directly in the `if`:
```suggestion
                    if ! ipmitool -I lanplus -U "${IPMI_USERNAME}" -P "${IPMI_PASSWORD}" -H "${TARGET_NCN_mgmt_host}" power off; then
```